### ISSUE TITLE
Abnormal RCNNL1Loss and MaskLogLoss after RuntimeWarning

### ISSUE THREAD

**Issue by realwecan** [TuSimple/mx-maskrcnn#16]
Hi guys,

I am encountering the following issue while training on a single TITAN X GPU:

INFO:root:Epoch[0] Batch [820]	Speed: 0.53 samples/sec	Train-RCNNAcc=0.753992,	RCNNLogLoss=0.927809,	RCNNL1Loss=4.257946,	MaskACC=0.902436,	MaskLogLoss=0.166465,	
/home/twang/work/fine_segmentation/baselines/mx-maskrcnn/rcnn/core/metric.py:134: **RuntimeWarning: invalid value encountered in greater**
  idx = np.where(np.logical_and(mask_prob > 0.5, mask_weight == 1))
INFO:root:Epoch[0] Batch [840]	Speed: 0.56 samples/sec	Train-RCNNAcc=0.753502,	RCNNLogLoss=0.987536,	RCNNL1Loss=24666331660818701805251597434880.000000,	MaskACC=0.901641,	MaskLogLoss=nan,	

**The RCNNL1Loss basically explodes after this RuntimeWarning.
The MaskLogLoss was okay before this RuntimeWarning, but becomes NaN afterwards.**

The error occurred during the 3rd training stage: **TRAIN RCNN WITH IMAGENET INIT AND RPN DETECTION**

Any advice will be much appreciated. Thanks!

**Comment by Zehaos** [comment]
Hi, @realwecan 
What default.base_lr are you use?

**Comment by realwecan** [comment]
Thanks @Zehaos for your prompt reply!

I did not make any changes to the default settings (e.g., I use the vanilla Cityscapes dataset with supplied hyperparameters). I checked the rcnn/config.py file, and the value is default.base_lr = 0.004.

Do you have any idea where the RuntimeWarning might be coming from? Thanks!

**Comment by winstywang** [comment]
Hi

  How many gpus do you use in training? Be aware of the relationship between learning rate ang effective batch size.

**Comment by realwecan** [comment]
@winstywang Thanks for bringing this up, I did use just one GPU. Should I maybe divide my base_lr by a factor of 4? Just wanted to know if there may be other causes for this RuntimeWarning.

**Comment by winstywang** [comment]
I think this should be the reason. The warning just means some value had become NaN at that time. Just have a try.

**Comment by realwecan** [comment]
Sure thanks a lot @winstywang!
